### ISSUE TITLE
Vimium corrupts Chrome XML view page

### ISSUE THREAD

**Issue by difanz** [philc/vimium#1640]
Running version 1.51 with Chrome Version 43.0.2357.37 beta (64-bit) on Mac OS X 10.10.3.

Example page: http://qz.com/feed/

Intended behavior: Show Chrome XML viewer for this page since it does not have any XSLT associated.

Actual behavior: Shows XML as well as text representation of Vimium CSS code.

Screenshot: http://tifan.la/7DpI

**Comment by smblott-github** [comment]
Thanks, @difanz.

And the Vomnibar fails to load...

![snapshot](https://cloud.githubusercontent.com/assets/2641335/7488959/e326f4ae-f3c8-11e4-87cf-ab7872b27f74.png)

Here...

![snapshot](https://cloud.githubusercontent.com/assets/2641335/7488969/044cdf54-f3c9-11e4-82fd-8400db66c872.png)

**Comment by mrmr1993** [comment]
Do our key handlers still work? If not, it makes sense to abort initialisation completely when `document.contentType == "application/rss+xml" `.

The only code I can see in the screenshot is from vimium.css. The only solutions I can think of are:
* inject the code when the `domReady` port is opened
  - this won't work with frames; we can't inject into specific frames.
* put all Vimium UI components in shadow DOMs (or all in one shadow DOM together) and only inject css into them (/it), not the document itself.

**Comment by smblott-github** [comment]
Our key handlers do work.  Except the Vomnibar, for which initialisation fails.

**Comment by mrmr1993** [comment]
How about the HUD?

**Comment by smblott-github** [comment]
HUD does not work.

**Comment by mrmr1993** [comment]
Okay, so we should block any attempts to create elements/add them to the DOM when we know they're going to fail. Easiest way should be to
* have a single shadow DOM that all our UI goes in, created at load;
  - (this also fixes the CSS issue)
* fail gracefully when creating the shadow DOM errors; and
* block commands that will try to inject UI from running if the shadow DOM errored/isn't available.
  - this should mean adding a one line check to `LinkHints.activateMode`, `Vomnibar.init`, `Vomnibar.activate`, and one or more HUD functions.

There are probably other ways to achieve this, but it's nice to have a single point of failure that we can check (the shadow DOM), and to avoid having to check mime types altogether.

**Comment by markstos** [comment]
"View Source" on an XML page in Chrome is also broken for me now. Content-Type is `application/xml`. Sample URL is private, sorry. Appears to the same issue.

**Comment by gdh1995** [comment]
I downloaded the .xml file, and open it in Chrome locally. Then when `F12`:
* there is a `<html>` tag
* `<style>` can be added into `<head>`, but its content is ignored (its CSS classes donot work);
* `<body>` and its children allows styles, which can be saw / changed in the right panel.
* `<div>` who is created by `document.createElement` has a `null` style

**Comment by mrmr1993** [comment]
Looks like I was wrong earlier. The text is coming from `UIComponent`. A quick fix is to change `UIComponent` not to inject the `<style>` element on XML pages. So

```coffee
shadowDOM.appendChild styleSheet
```

becomes

```coffee
shadowDOM.appendChild styleSheet if styleSheet.style?
```

with some comment explaining why.

To fix the `HUD` error, we could move over to using `HUD._displayElement.classList.add`/`.remove` and put the styles in our injected stylesheet.

**Comment by winny-** [comment]
:+1: Is this waiting on a new release?

**Comment by mrmr1993** [comment]
@winny- unfortunately, yes.

**Comment by Falconne** [comment]
It's been a while since this was fixed, is there a new release any time soon?

**Comment by smblott-github** [comment]
@Falconne. Not sure.  Next month or so, probably.

**Comment by smblott-github** [comment]
Closing.
This is fixed in master and will be in 1.52 (which will be released at some as-yet unknown point).

**Comment by rrsahuone** [comment]
The best aspect is that everything can be shifted promptly without any significant or minor reduction. Movers and packers are using high quality content to package fragile components like cotton and others. 

<a href="http://www.expert5th.in/packers-and-movers-hyderabad/" title="Packers and Movers in Hyderabad">Packers and Movers in Hyderabad</a>
<a href="http://www.expert5th.in/packers-and-movers-bangalore/" title="Packers and Movers in Bangalore">Packers and Movers in Bangalore</a>
<a href="http://www.expert5th.in/packers-and-movers-mumbai/" title="Packers and Movers in Mumbai">Packers and Movers in Mumbai</a>

**Comment by rrsahuone** [comment]
They are also providing solutions across Bangalore , Bangalore all close by locations especially NCR Professional.

<a href="http://www.expert5th.in/packers-and-movers-gurgaon/" title="Packers and Movers in Gurgaon">Packers and Movers in Gurgaon</a>
<a href="http://www.expert5th.in/packers-and-movers-delhi/" title="Packers and Movers in Delhi">Packers and Movers in Delhi</a>
<a href="http://www.expert5th.in/packers-and-movers-pune/" title="Packers and Movers in Pune">Packers and Movers in Pune</a>

**Comment by rrsahuone** [comment]
Obviously expenses will vary depending on place, range and baggage you wanted to transportation from one position to other. 
<a href="http://www.expert5th.in/packers-and-movers-chennai/" title="Packers and Movers in Chennai">Packers and Movers in Chennai</a>
<a href="http://www.expert5th.in/packers-and-movers-noida/" title="Packers and Movers in Noida">Packers and Movers in Noida</a>
<a href="http://www.expert5th.in/packers-and-movers-navimumbai/" title="Packers and Movers in Navi Mumbai">Packers and Movers in Navi Mumbai</a>
<a href="http://www.expert5th.in/packers-and-movers-thane/" title="Packers and Movers in Thane">Packers and Movers in Thane</a>
<a href="http://www.expert5th.in/packers-and-movers-ghaziabad/" title="Packers and Movers in Ghaziabad">Packers and Movers in Ghaziabad</a>
<a href="http://www.expert5th.in/packers-and-movers-faridabad/" title="Packers and Movers in Faridabad">Packers and Movers in Faridabad</a>

**Comment by rrsahuone** [comment]
We help in moving electronic products as well like freeze, TV, washing machine, microwave, and Air Conditioner. 

<a href="http://www.expert5th.in/packers-and-movers-bangalore/" title="Packers and Movers Bangalore">Packers and Movers Bangalore</a>
<a href="http://www.expert5th.in/packers-and-movers-pune/" title="Packers and Movers Pune">Packers and Movers Pune</a>
<a href="http://www.expert5th.in/packers-and-movers-mumbai/" title="Packers and Movers Mumbai">Packers and Movers Mumbai</a>
